charms/worker/k8s/src/token_distributor.py

```diff
@@ -193,6 +193,30 @@ def _revoke_cos_token(self, name: str, _):
         """
         # TODO: implement removing cos token
 
+    def _get_juju_secret(self, relation: ops.Relation, unit: ops.Unit) -> Optional[str]:
+        """Lookup juju secret offered to a unit on this relation.
+
+        Args:
+            relation (ops.Relation): Which relation (cluster or k8s-cluster)
+            unit (ops.Unit): The unit the secret is intended for
+
+        Returns:
+            secret_id (None | str) if on the relation
+        """
+        return relation.data[self.charm.unit].get(SECRET_ID.format(unit.name))
+
+    def _revoke_juju_secret(self, relation: ops.Relation, unit: ops.Unit) -> None:
+        """Revoke and remove juju secret offered to a unit on this relation.
+
+        Args:
+            relation (ops.Relation): Which relation (cluster or k8s-cluster)
+            unit (ops.Unit): The unit the secret is intended for
+        """
+        secret_id = SECRET_ID.format(unit.name)
+        if juju_secret := relation.data[self.charm.unit].pop(secret_id, None):
+            secret = self.charm.model.get_secret(id=juju_secret)
+            secret.remove_all_revisions()
+
```

Comment:
These are here to clean up the tokens we keep making that juju is holding onto forever.  Once a token is USED (or revoked) juju doesn't need to know about that secret anymore.  Let's be a good citizen and tidy up